lib/util/http/host-rules.ts

```diff
@@ -162,6 +162,13 @@ export function applyHostRule<GotOptions extends HostRulesGotOptions>(
     options.lookup = dnsLookup;
   }
 
+  if (hostRule.headers) {
+    options.headers = {
+      ...options.headers,
+      ...hostRule.headers,
```

Comment:
We need to think if there are any malicious or nuisance capabilities we need to protect against.

For example, what if someone overrode the user agent of the hosted app to include vulgarities or death threats against GitHub or others? In that case potentially the entire app gets blocked.

I'd like to avoid an allowList concept but sometimes that's unfortunately the only way.

Reply:
I think we can add a global option (glob/regex) to limit headers?

Reply:
Hi both, thanks for checking this PR.
One safe solution would be to add this explicit `X-Auth-Token` field. But, it would be an endless issue as long as a user needs to use a new custom header. Moreover, struct would explode. I don't believe having a long list of static fields is a good solution.
On the other hand, a free headers field for user solves many issues but increases security issue if all headers are allowed to be overwritten. What about beginning with a more or less strict regex like enabling all fields beginning with `X-` ?
I believe it would already solves many cases. And consider open "the door" (improve regex) based on user request ? (PR would be small)
Thanks

Reply:
How about this?
- We add a new field `allowedUserHeaders` which takes an array of regex patterns
- We have some sensible defaults, such as `X-` prefixes, and add to it over time
- Admins of self-hosted can supplement it through config at any time they want

Reply:
I have pushed some changes, could you confirm that it matches what you were expecting? If yes, I will write the documentation and tests.

Reply:
I have pushed some changes, could you confirm that it matches what you were expecting? If yes, I will write the documentation and tests.